### PR TITLE
Fixed issue with having multiple includeWith stanzas in different extensions

### PR DESCRIPTION
**Pull request by iuyiuy**

### CODE REVIEW

**Comment by codedread** [comment]
Sorry I didn't understand this change - can you clarify a little more what it does?

**Comment by iuyiuy** [comment]
In extensions/ext-connector.js there's a button specified with an includeWith object, which defines that it should be included in the same flyout menu as the line tool. When i tried to make an extension, which included another button in the same menu, it didn't work. These changes made that work.

File: editor/svg-editor.js [hunk 1]
```diff
@@ -2037,20 +2041,6 @@ TODOS
...
-									$.each(holders[opts.parent], function(i, tool) {
-										if (tool.sel == currentOperation) {
-											if (!event.shiftKey || !flyoutIsSelected) {
-												options = tool;
-											} else {
-												options = holders[opts.parent][i+1] || holders[opts.parent][0];
-											}
-										}
-									});
-								}
```
Comment (codedread) [inline editor/svg-editor.js]: Why is this code being removed?

File: editor/svg-editor.js [hunk 2]
```diff
@@ -2019,14 +2019,18 @@ TODOS
...
 					var shower = $(show_sel);
 					var def = false;
-					buttons.addClass('tool_button')
+					buttons.addClass('tool_button tool_button_evt_handled')
 						.unbind('click mousedown mouseup') // may not be necessary
-						.each(function(i) {
+						.each(function() {
 							// Get this buttons options
+							var i;
+							for(var j in btn_opts)
```
Comment (codedread) [inline editor/svg-editor.js]: Also, you should break after i = j, correct?

File: editor/svg-editor.js (hunk 1 above)
Comment (iuyiuy) [inline editor/svg-editor.js]: My commit message for that change said "Removed some code that would select the wrong tool when using keyboard shortcuts for flyout buttons." On the svg-edit demo page you might notice that the keyboard shortcut for the line tool [L] doesn't work properly:
https://svg-edit.github.io/svgedit/releases/svg-edit-2.8.1/svg-editor.html

I haven't gone back and tested it now, but i believe this code was part of the issue and didn't seem to serve a purpose. I haven't been able to find a keyboard shortcut that doesn't work as expected after this change.

File: editor/svg-editor.js (hunk 2 above)
Comment (iuyiuy) [inline editor/svg-editor.js]: Of course. Thanks for spotting that. These changes are now committed and pushed.

File: editor/svg-editor.js (hunk 1 above)
Comment (codedread) [inline editor/svg-editor.js]: Ok, I played around with it on the trunk and it seems that pressing L selects the line tool but it doesn't show up in the toolbar as selected, while pressing R selects the rectangle tool and it shows as selected.  I guess you are reasonably sure this doesn't break anything so let's go for it.

File: editor/svg-editor.js [hunk 3]
```diff
@@ -2029,8 +2029,11 @@ TODOS
 						.each(function() {
 							// Get this buttons options
 							var i;
-							for(var j in btn_opts)
-								if(btn_opts[j].sel == '#'+this.getAttribute('id')) i = j;
+							for(var j in btn_opts) {
+								if(btn_opts[j].sel == '#'+this.getAttribute('id')) {
```
Comment (codedread) [inline editor/svg-editor.js]: And a space after "for" and before (

**Comment by brettz9** [comment]
Fix applied in `master` just now (https://github.com/brettz9/svgedit/commit/0a2da503f1136bff67bf884fd01d1cf4abae0800) which incorporates this PR. I included the `keypress` code portion, however, albeit with adjustments, as that was serving the purpose of allowing `shift + <key>` to cycle through the flyouts. Closing therefore but feel free to comment further.

**Pull request closed without merging**